### PR TITLE
Dto

### PR DESCRIPTION
**Pull request by ssssarah**

### CODE REVIEW

File: src/main/compa/controllers/FriendshipController.java [hunk 1]
```diff
@@ -1,143 +1,125 @@
...
+    public FriendshipController(Container container) {
+        super(PREFIX, container);
+        this.registerAuthRoute(HttpMethod.POST, "/request", this::requestFriendship, "application/json");
+        this.registerAuthRoute(HttpMethod.POST, "/response", this::respondToRequest, "application/json");
+        this.registerAuthRoute(HttpMethod.GET, "/friend_list", this::getFriends, "application/json");
+        this.registerAuthRoute(HttpMethod.GET, "/pending", this::getPendingFriendships, "application/json");
 
-        friendshipDAO = (FriendshipDAO) modelManager.getDAO(Friendship.class);
-        userDAO = (UserDAO) modelManager.getDAO(User.class);
+        friendshipDAO = (FriendshipDAO) container.getDAO(Friendship.class);
```
Comment (giardiv) [inline src/main/compa/controllers/FriendshipController.java]: @ssssarah we can access to the container from a controller subclass ?  It sounds dangerous ☄️ Container should be private inside the Controller superclass, no ?

File: src/main/compa/controllers/FriendshipController.java [hunk 2]
```diff
@@ -1,143 +1,125 @@
...
+        userDAO.findById(routingContext.request().getParam("friend_id"), res1 -> {
 
-        if(me == null){
-            routingContext.response().end("can't find authenticated user"); //TODO FORMAT
-            return;
-        }
+            User friend = res1.result();
+
+            if(friend == null){
+                routingContext.response().end("can't find your friend"); //TODO FORMAT
```
Comment (giardiv) [inline src/main/compa/controllers/FriendshipController.java]: @ssssarah the formatting of the answer could be done from the `checkParams` method

File: src/main/compa/controllers/FriendshipController.java [hunk 3]
```diff
@@ -1,143 +1,125 @@
...
-        }
+            friendshipDAO.addFriendship(me, friend, res2 -> {
+                if(res2.failed()){
+                    routingContext.response().setStatusCode(418).end(gson.toJson(res2.cause()));
+                }
+                else{
+                    routingContext.response().end("you are now friends"); //TODO FORMAT
+                }
+            });
+        });
```
Comment (giardiv) [inline src/main/compa/controllers/FriendshipController.java]: Wonderful job @ssssarah 👏

File: src/main/compa/daos/UserDAO.java [hunk 4]
```diff
@@ -1,57 +1,72 @@
...
+
     }
 
-    @Override
-    public User findById(String id) {
-        logger.log(Level.INFO, "Looking for user {0}", id);
-        User u = super.findById(id);
-        logger.log(Level.INFO, "User {0} found", u == null ? "not" : "");
-        return u;
+    public void findById(String id, Handler<AsyncResult<User>> resultHandler) {
```
Comment (giardiv) [inline src/main/compa/daos/UserDAO.java]: @ssssarah Is it possible to delegate this async job to the DAO ? I mean, the call of this `vertx.executeBlocking` is recurrent and it's called on each method. Shouldn't we call a generic function which will run our blocking method ?

File: src/main/compa/services/AuthenticationService.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+            ((UserDAO) container.getDAO(User.class)).findOne("token", token, res -> {
+                User u = res.result();
+                Future<User> f = (res.result() == null)
+                        ? Future.failedFuture("no user found with this token")
+                        : Future.succeededFuture(u);
+                f.setHandler(resultHandler);
+            });
+        };
+
```
Comment (giardiv) [inline src/main/compa/services/AuthenticationService.java]: Nice 🙆‍♂️

File: src/main/compa/services/GsonService.java [hunk 6]
```diff
@@ -1,41 +1,21 @@
...
-    {
-        return INSTANCE;
-    }
-
-    public GsonBuilder getGsonBuilder(){
-        return this.gb;
+    public String toJson(Object thing){
+        return gson.toJson(thing);
     }
 }
```
Comment (giardiv) [inline src/main/compa/services/GsonService.java]: @ssssarah I don't like idea that every services are "container aware" 🧐

File: src/main/compa/controllers/FriendshipController.java (hunk 1 above)
Comment (ssssarah) [reply src/main/compa/controllers/FriendshipController.java]: it is, but in order to call the super method, we need the container as a parameter of the constructor. Therefore, the container can be accessed inside the constructor, but not outside (in any other method of the class)

File: src/main/compa/controllers/FriendshipController.java (hunk 2 above)
Comment (ssssarah) [reply src/main/compa/controllers/FriendshipController.java]: check params only checks whether the request contains the parameters we need. 
Here, we're in a situation where the parameter does exist, but when we use it to find the user instance, we find nothing. (the id is not valid)

File: src/main/compa/controllers/FriendshipController.java (hunk 3 above)
Comment (ssssarah) [reply src/main/compa/controllers/FriendshipController.java]: thanks :)

File: src/main/compa/services/GsonService.java (hunk 6 above)
Comment (ssssarah) [reply src/main/compa/services/GsonService.java]: you'll have access to the container inside the constructor, yes. But later on, it'll be stored in a private field of the superclass, so you can limit the service's access to the container by using limiting getters (that target only one resource of the container)

File: src/main/compa/services/AuthenticationService.java (hunk 5 above)
Comment (ssssarah) [reply src/main/compa/services/AuthenticationService.java]: :+1:

File: src/main/compa/daos/UserDAO.java (hunk 4 above)
Comment (ssssarah) [reply src/main/compa/daos/UserDAO.java]: I like the idea, but I'd have to look into it because I don't have an immediate answer on how to implement that

**Pull request merged**